src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.cs

```diff
@@ -184,5 +185,33 @@ protected override void GetUnusedNamespaceImports(SemanticModel model, HashSet<S
                 }
             }
         }
+
+        internal static SyntaxToken ExtractAnonymousTypeMemberName(ExpressionSyntax input)
+        {
+            while (true)
+            {
+                switch (input.Kind())
+                {
+                    case SyntaxKind.IdentifierName:
+                        return ((IdentifierNameSyntax)input).Identifier;
+
+                    case SyntaxKind.SimpleMemberAccessExpression:
+                        input = ((MemberAccessExpressionSyntax)input).Name;
+                        continue;
+
+                    case SyntaxKind.ConditionalAccessExpression:
+                        input = ((ConditionalAccessExpressionSyntax)input).WhenNotNull;
+                        if (input.Kind() == SyntaxKind.MemberBindingExpression)
```

Comment:
i always get confused by conditional access syntax. waht happens if you have "x?.y.z"?  is ".y.z" a member binding?  or it is a member access?

(please add test regardless).